data.js

```diff
@@ -0,0 +1,62 @@
+const popura = require('popura');
+const _ = require('lodash');
+ 
+class AbstractAnimeDataSource {
+    getAnime(name) {
+        throw "Implementation error";
+    }
+
+    _constructAnimeModel(animeResponse) {
+        throw "Implementation error";
+    }
+}
+
+class Anime {
+    constructor(params) {
+        this._validateParams(params);
```

Comment:
It might be worth making an abstract model class with `_validateParams` function signature